### PR TITLE
Pass sysinfo opts as comma separated values

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit fixes the following:
when default is passed as a syinfo option, pbench-sysinfo-dump
collects just block info since it finds space separated values
instead of comma separated values.

### CODE REVIEW

**Comment by chaitanyaenr** [comment]
Fixes #656

File: agent/util-scripts/pbench-collect-sysinfo [hunk 1]
```diff
@@ -99,7 +99,7 @@ if $check; then
 		:
 	else
 		for item in ${sysinfo//,/ }; do
-			if echo "${sysinfo_opts[@]}" | grep -q -w "$item"; then
+			if echo "${sysinfo_opts_default[@]}" | grep -q -w "$item"; then
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Why did you change this? Shouldn't you be checking the actual sysinfo opts, not the default ones?

File: agent/util-scripts/pbench-collect-sysinfo [hunk 2]
```diff
@@ -116,11 +116,8 @@ fi
 if [ "$sysinfo" == "none" ]; then
 	exit 0
 fi
-if [ "$sysinfo" == "default" ]; then
-	sysinfo=${sysinfo_opts_default[*]}
-fi
-if [ "$sysinfo" == "all" ]; then
-	sysinfo=${sysinfo_opts[*]}
+if [ "$sysinfo" == "default" ] || [ "$sysinfo" == "all" ]; then
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Why are there three variables? sysinfo_default_opts, sysinfo_opts and sysinfo - what do they do?

File: agent/util-scripts/pbench-collect-sysinfo [hunk 3]
```diff
@@ -17,7 +17,7 @@ pbench_bin="`cd ${script_path}/..; /bin/pwd`"
 group=default
 dir="/tmp"
 sysinfo_opts_default=( block libvirt kernel_config sos topology ara )
-sysinfo_opts=${sysinfo_opts_default[*]}
+sysinfo_opts=$(IFS=,; echo "${sysinfo_opts_default[*]}")
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Is the only difference between these two that one is space-separated and the other is comma separated? Do they otherwise always contain exactly the same elements?

Comment (chaitanyaenr) [reply agent/util-scripts/pbench-collect-sysinfo]: yes, the only difference between two is that sysinfo_opts_default is an array with all the options while sysinfo_opts contains space separated values and they always contain exactly the same elements.

File: agent/util-scripts/pbench-collect-sysinfo (hunk 1 above)
Comment (chaitanyaenr) [reply agent/util-scripts/pbench-collect-sysinfo]: sysinfo has the actual passed sysinfo opts, we are checking if the passed opts are valid or not.

File: agent/util-scripts/pbench-collect-sysinfo (hunk 2 above)
Comment (chaitanyaenr) [reply agent/util-scripts/pbench-collect-sysinfo]: sysinfo is set with the parameters passed by the user, sysinfo_default_opts is an array of all possible options while sysinfo_opts contains comma separated values.

**Comment by ndokos** [comment]
Why is `pbench-collect-sysinfo --dir foo beg|end` not equivalent to  `pbench-collect-sysinfo --dir foo beg|end --sysinfo=default`? I see that `pbench-fio`, `pbench-uperf`, `pbench-moongen` and `pbench-user-benchmark` have been modified to pass `--sysinfo=..` to `pbench-collect-sysinfo`, but all the others do not. They all need to be modified so that they recognize the switch, but at least for now, interpreting the absence of a --sysinfo options as requiring the default, would allow them all to collect data, rather than punting.

File: agent/util-scripts/pbench-collect-sysinfo (hunk 2 above)
Comment (ndokos) [reply agent/util-scripts/pbench-collect-sysinfo]: Do you need sysinfo_opts at all? If you can drop it easily, I'd suggest you do so. If not, then it should be renamed.

File: agent/util-scripts/pbench-collect-sysinfo [hunk 4]
```diff
@@ -16,8 +16,10 @@ pbench_bin="`cd ${script_path}/..; /bin/pwd`"
 # Defaults
 group=default
 dir="/tmp"
+# array containing all the possible sysinfo options
 sysinfo_opts_default=( block libvirt kernel_config sos topology ara )
-sysinfo_opts=${sysinfo_opts_default[*]}
+# get comma seperated values
+sysinfo_comma_seperated_opts_default=$(IFS=,; echo "${sysinfo_opts_default[*]}")
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Spelling error: "separated"

You should call it either "comma_separated_sysinfo_opts_default" or "sysinfo_opts_default_comma_separated" to emphasize the relation between the two variables. I prefer the second one btw.

Comment (chaitanyaenr) [reply agent/util-scripts/pbench-collect-sysinfo]: Fixed.

File: agent/util-scripts/pbench-collect-sysinfo [hunk 5]
```diff
@@ -16,14 +16,16 @@ pbench_bin="`cd ${script_path}/..; /bin/pwd`"
 # Defaults
 group=default
 dir="/tmp"
+# array containing all the possible sysinfo options
 sysinfo_opts_default=( block libvirt kernel_config sos topology ara )
-sysinfo_opts=${sysinfo_opts_default[*]}
+# get comma seperated values
```
Comment (ndokos) [inline agent/util-scripts/pbench-collect-sysinfo]: Spelling: "separated"
